### PR TITLE
[DO NOT MERGE] [DEVELOPER-5868] Update Assembly, Compose and some patches

### PR DESCRIPTION
**Pull request by jordanpagewhite**
This updates the Assembly and Compose modules to their latest versions,
and it provides some new patches and an update to an existing core
patch.

### JIRA Issue Link
* This is not specific to a single issue, but it includes all of the changes for the Authoring Experience re-work

See https://issues.jboss.org/browse/DEVELOPER-5868 for the epic

### Known Issue(s)

* [Compose can break modals if modal height extends beyond browser height](https://issues.jboss.org/browse/DEVELOPER-5912)

### Verification Process

* TBD

* Compose is at b03cb77a
* Assembly is at 5cef2ae5
* We've applied the Bootstrap patch "Remove overflow style from modal": "https://www.drupal.org/files/issues/2019-05-24/modal-height-3048592-4.patch"
* We have updated the Drupal core patch to add disabled layout to "Add disabled layout": "https://www.drupal.org/files/issues/2019-05-23/core-add_disabled_layout-3056712-1-8.7.2.patch"

### CODE REVIEW

**Comment by jordanpagewhite** [comment]
If you use the Compose Widget on the Sections field of the Page content type and try to add new content, then you will receive the following error:

```
TypeError: Argument 1 passed to Drupal\compose\EntityComposeWidget::addExistingRow() must be an instance of Drupal\Core\Entity\EntityBase, instance of Drupal\assembly\Entity\Assembly given, called in /var/www/drupal/web/modules/contrib/compose/src/Plugin/Field/FieldWidget/ComposeWidget.php on line 650 in Drupal\compose\EntityComposeWidget->addExistingRow() (line 133 of /var/www/drupal/web/modules/contrib/compose/src/EntityComposeWidget.php) #0 
```

I think that we just need to change the type hinting from EntityBase to ContentEntityBase. I tried it locally, and that resolved the error. I believe that ContentEntityBase will still provide support for all the entity types we plan on supporting with this widget.

**Comment by jordanpagewhite** [comment]
I've opened up a PR against Alloy's Compose gitlab here to resolve the typehinting error: https://code.kilpho.st/alloy/compose/merge_requests/19

After making that change locally, I was able to successfully add new and existing references.

File: _docker/drupal/drupal-filesystem/composer.json [hunk 1]
```diff
@@ -60,7 +59,7 @@
         "acquia/lightning": "~3.2.6",
         "ckeditor/indentblock": "4.11.4",
         "drupal/admin_toolbar": "1.26",
-        "drupal/assembly": "1.x-dev#94645cf6617d5f375ba2cc3c060df13c88e75d8b",
+        "drupal/assembly": "1.x-dev",
```
Comment (robpblake) [inline _docker/drupal/drupal-filesystem/composer.json]: @jordanpagewhite I assume this leaves the potential for this dependency to change between builds. Can we lock it to a commit that you're happy with please?

File: _docker/drupal/drupal-filesystem/composer.json [hunk 2]
```diff
@@ -101,7 +100,7 @@
         "drupal/akamai": "3.0-alpha4",
         "drupal/trailing_slash": "^1.0",
         "drupal/bootstrap": "3.17",
-        "drupal/compose": "1.x",
+        "drupal/compose": "1.x-dev",
```
Comment (robpblake) [inline _docker/drupal/drupal-filesystem/composer.json]: @jordanpagewhite Same comment as above, can we please lock the dependency to a known good commit in the source repo.

cheers,

Rob

**Comment by robpblake** [comment]
@jordanpagewhite Can you also please update `composer.json` and `composer.lock` to include the changes related to your user roles work which I have just merged.

cheers,

Rob

**Comment by robpblake** [comment]
> If you use the Compose Widget on the Sections field of the Page content type and try to add new content, then you will receive the following error:
> 
> ```
> TypeError: Argument 1 passed to Drupal\compose\EntityComposeWidget::addExistingRow() must be an instance of Drupal\Core\Entity\EntityBase, instance of Drupal\assembly\Entity\Assembly given, called in /var/www/drupal/web/modules/contrib/compose/src/Plugin/Field/FieldWidget/ComposeWidget.php on line 650 in Drupal\compose\EntityComposeWidget->addExistingRow() (line 133 of /var/www/drupal/web/modules/contrib/compose/src/EntityComposeWidget.php) #0 
> ```
> 
> I think that we just need to change the type hinting from EntityBase to ContentEntityBase. I tried it locally, and that resolved the error. I believe that ContentEntityBase will still provide support for all the entity types we plan on supporting with this widget.

@jordanpagewhite Does this block this PR going in?

**Comment by jordanpagewhite** [comment]
@robpblake Yes, that issue blocks this PR. I am discussing this issue with Matt and Roger right now to get a consensus on how we would like to resolve this issue. I'll update the PR shortly.

File: _docker/drupal/drupal-filesystem/composer.json (hunk 2 above)
Comment (jordanpagewhite) [reply _docker/drupal/drupal-filesystem/composer.json]: @robpblake I resolved this in the commit I just pushed. Thanks!

File: _docker/drupal/drupal-filesystem/composer.json (hunk 1 above)
Comment (jordanpagewhite) [reply _docker/drupal/drupal-filesystem/composer.json]: @robpblake I resolved this in the commit I just pushed. Thanks!

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by robpblake** [comment]
@jordanpagewhite Is there also a discussion to be had here around the initial performance metrics we've seen from `compose` on New Relic before we add even more functionality?

**Comment by robpblake** [comment]
@jordanpagewhite This will need updating to take into account your 8.7.3 core update.

cheers,

Rob

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by robpblake** [comment]
retest this please

**Comment by robpblake** [comment]
rebuild this please

**Comment by robpblake** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by robpblake** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by fearlsgroove** [comment]
rebuild this please

**Comment by robpblake** [comment]
@jordanpagewhite I'm closing this to reclaim some resources from our PR environment quota, particularly given #3300 . Please just re-open if you need too.

cheers,

Rob

**Pull request closed without merging**